tests/Doctrine/Tests/ORM/Functional/Ticket/GH7864Test.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Tests\Models\Tweet\Tweet;
+use Doctrine\Tests\Models\Tweet\User;
+use Doctrine\Tests\OrmFunctionalTestCase;
+use function array_values;
+
+class GH7864Test extends OrmFunctionalTestCase
+{
+    protected function setUp() : void
+    {
+        $this->useModelSet('tweet');
+
+        parent::setup();
+    }
+
+    public function testExtraLazyRemoveElement()
+    {
+        $user       = new User();
+        $user->name = 'test';
+
+        $tweet1          = new Tweet();
+        $tweet1->content = 'Hello World!';
+        $user->addTweet($tweet1);
+
+        $tweet2          = new Tweet();
+        $tweet2->content = 'Goodbye, and thanks for all the fish';
+        $user->addTweet($tweet2);
+
+        $this->_em->persist($user);
+        $this->_em->persist($tweet1);
+        $this->_em->persist($tweet2);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $user  = $this->_em->find(User::class, $user->id);
+        $tweet = $this->_em->find(Tweet::class, $tweet1->id);
+
+        $user->tweets->removeElement($tweet);
+
+        $tweets = $user->tweets->map(static function (Tweet $tweet) {
+            return $tweet->content;
+        });
+
+        $this->assertEquals(['Goodbye, and thanks for all the fish'], array_values($tweets->toArray()));
```

Comment:
Please use `self::assert*` since these methods are static.

Reply:
The tests all use $this->assert* at the moment. Shouldn't we enforce this via coding standard instead so that we have a phpcbf fix for it instead of requiring us to do this kind of remark manually?